### PR TITLE
Add lowering for adaptive avg pool

### PR DESCRIPTION
**Pull request by eellison**
The two occurrences in torchbench of this op are both no-ops lol. When you disable the no-op optimization, the op is ~6x faster (again, for those two sets of inputs).

### CODE REVIEW

**Comment by ngimel** [comment]
Hm, I would think they reduce to 1x1 output and thus could be replaced by sum, not no-op?

File: torchinductor/lowering.py [hunk 1]
```diff
@@ -2123,6 +2125,107 @@ def fn(idx):
...
+
+    h_in = V.graph.sizevars.guard_static_shape(h_in)
+    w_in = V.graph.sizevars.guard_static_shape(w_in)
+
+    h_out, w_out = output_size
+
+    # no-op if the same input and output
+    if h_in == h_out and w_in == w_out:
+        # TODO: do I need to copy ? _to_copy does not
+        return x
```
Comment (ngimel) [inline torchinductor/lowering.py:2173]: better do a copy, yes (to make sure that previously run functionalization is still valid)

**Comment by eellison** [comment]
```
>>> x = torch.rand(1, 1, 4, 4)
>>> y = torch.ops.aten._adaptive_avg_pool2d(x, [4, 4])
>>> (x - y).nonzero().shape
torch.Size([0, 4])
```

The output size is given, not the kernel size which differs from the other apis... When the output_size == input_size for adaptive average pooling, that means the kernel size will be [1, 1] and you're doing a 1-element average (no-op).

File: torchinductor/lowering.py (hunk 1 above)
Comment (jansel) [reply torchinductor/lowering.py:2173]: https://github.com/pytorch/torchdynamo/blob/4b212e1dce0431a2903426dcd57369eb85ff47d5/torchinductor/lowering.py#L915

**Review by jansel — COMMENTED** [review]
We should add a test for this.

Kind of funny these are no-ops, though not the first time I've seen cases like that.

**Review by jansel — APPROVED** [review]

**Pull request merged**